hedera-node/src/main/java/com/hedera/services/tokens/HederaTokenStore.java

```diff
@@ -454,6 +452,20 @@ public TokenCreationResult createProvisionally(TokenCreateTransactionBody reques
 		return success(pendingId);
 	}
 
+	public void addKnownTreasury(AccountID aId, TokenID tId) {
+		if (!knownTreasuries.containsKey(aId)) {
+			knownTreasuries.put(aId, new HashSet<>());
+		}
+		knownTreasuries.get(aId).add(tId);
+	}
+
+	public void removeKnownTreasuryForToken(AccountID aId, TokenID tId) {
+		knownTreasuries.get(aId).remove(tId);
```

Comment:
I would prefer if we have a check that validates that `aId` is present in the map similarly to the `get(TokenID id)` function